### PR TITLE
Add mesh polygonization and getFaceFaceTableNonManifold()

### PR DESCRIPTION
**Pull request by tomas2211**
- Mesh polygonization `polygonizeMesh()` - merges triangles in one region to polygons and creates polygonal mesh
- `saveAsObj()` for `MultiPolyMesh`
- `getFaceFaceTableNonManifold()` - separate function to get face adjacency map for non-manifolds - implied refactor of `removeNonManifoldEdges()`

### CODE REVIEW

File: geometry/meshop.hpp [hunk 1]
```diff
@@ -227,6 +228,22 @@ Mesh::pointer refine( const Mesh &mesh, unsigned int maxFacesCount);
...
+using FaceFaceTable = std::vector<boost::container::small_vector<std::size_t, 3>>;
+
+/** Computes face incidency map, supports non manifold edges (face may have >3
+ * neighbors).
+ *
+ * NB: when two faces are incident over multiple edges, they are repated
+ * respective amount of times
+ *
+ * @param[in] mesh mesh to process
+ * @returns face incicency table
```
Comment (susterdan) [inline geometry/meshop.hpp:243]:
```suggestion
 * @returns face incidency table
```

File: geometry/meshop.hpp [hunk 2]
```diff
@@ -227,6 +228,22 @@ Mesh::pointer refine( const Mesh &mesh, unsigned int maxFacesCount);
...
 
+/**
+ * Face incidency - for each face lists all neighbors, supports non-manifolds
+ */
+using FaceFaceTable = std::vector<boost::container::small_vector<std::size_t, 3>>;
+
+/** Computes face incidency map, supports non manifold edges (face may have >3
+ * neighbors).
+ *
+ * NB: when two faces are incident over multiple edges, they are repated
```
Comment (susterdan) [inline geometry/meshop.hpp:239]:
```suggestion
 * NB: when two faces are incident over multiple edges, they are repeated
```

File: geometry/meshop.hpp (hunk 1 above)
Comment (tomas2211) [reply geometry/meshop.hpp]: Oops, fixed

**Review by susterdan — APPROVED** [review]
Sorry forgot to approve. Code looks good to me.

**Pull request merged**